### DISCUSSION TITLE
Can't host servers

### DISCUSSION THREAD

**Discussion by n00b660** [FakeFishGames/Barotrauma#13946] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When attempting to host server the popups shown in the screenshot appear but if i try to host DedicatedServer.exe it works perfectly fine, no mods are installed and joining servers is also working. 
<img width="301" alt="image" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/24531c30-363e-4318-b32b-52cbec6d0877">
<img width="292" alt="image2" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/256e452b-20ea-4a84-9e1c-3dd26673f8ec">
<img width="325" alt="image3" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/7fe10387-c1fe-424d-9cc7-d73016916299">
<img width="671" alt="image4" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/c6d643ad-1178-4695-b6e8-cbff46f0ca3a">
<img width="660" alt="image5" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/edd96d41-13b4-487d-bca4-6646d792974b">

I tried verifying files, reinstalling game, removing everything barotrauma related from appdata, unsubscribing from all mods but that didn't fix the issue.

### Reproduction steps

1. Attempt to host server 2. Get server crash popup

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
There aren't any crash reports
```

**Comment by n00b660** [comment]
Please help

**Comment by n00b660** [comment]
I still can't host servers

**Comment by n00b660** [comment]
still can't host... this sucks please help

**Reply by SomeRandomNoobKekeke** [reply]
Are you sure there's no servercrashreport.log in barotrauma folder?

**Reply by n00b660** [reply]
absolutely nothing

**Reply by n00b660** [reply]
I also tried to get help in discord server but nothing helped https://discord.com/channels/103209684680323072/587647684794515468/1237782362025230416

**Reply by SomeRandomNoobKekeke** [reply]
and no servercrashreportslim.log?

**Comment by SomeRandomNoobKekeke** [comment]
Are you on windows? could you go [here](https://github.com/evilfactory/LuaCsForBarotrauma/releases/tag/latest) 
Download client and server builds for your system (e.g. luacsforbarotrauma_build_windows_client.zip and luacsforbarotrauma_build_windows_server.zip)
create a new folder and unpack both arhives there
Then go to barotrauma folder (e.g. C:\Program Files (x86)\Steam\steamapps\common\Barotrauma) And copy Content folder from there to your new folder
Run barotrauma.exe and try again? will it crash?

**Reply by n00b660** [reply]
I will try this when I am home, thanks!

**Reply by n00b660** [reply]
<img width="946" alt="image" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/2ba9c901-bc64-411e-a3ee-cf836faa9fa6">
still crashes....

**Reply by SomeRandomNoobKekeke** [reply]
Then idk
![barodev small](https://github.com/FakeFishGames/Barotrauma/assets/122838333/b4a03033-43cc-4a96-ae67-d3d9fa4177ff)
![Homer cooking](https://media.tenor.com/v5hgv1Asw4sAAAAC/homer-cooking.gif)

**Reply by SomeRandomNoobKekeke** [reply]
So... There's nothing red in console when you try to host, no servercrashreport.log or servercrashreportslim.log in game folder?

**Reply by n00b660** [reply]
nothing

**Reply by SomeRandomNoobKekeke** [reply]
Then excuse me for not helping
![barodev small](https://github.com/FakeFishGames/Barotrauma/assets/122838333/7fa4bd19-80e2-43e0-ae88-485a1254b722)
I thought i could make a build where server send you some useful info instead of "oops, i crashed", but it turned out, those are not messages from server, and there's no way to know

I managed to crash the server on startup, I got the same messages as yours, but my server wrote me servercrashreport.log before it died
Think, what could prevent game from writing a file to that folder? Mb you got some weird permissions, idk

Also, when server dies, it tries to 1 save crash report 2 send report via gameAnalytics 3 say bb to clients
Enable gameAnalytics if you didn't, mb then some crash reports actually reach the barodevs and they'll notice you

**Comment by BlackMoons** [comment]
The error claims it could not find serversettings.xml at steam\common\barotrauma

make sure it exists. You could try verifying files as I believe that is part of the default barotrauma install. (you'll have to reinstall LUA afterwards)

Also make sure you have some free disk space, as I can't imagine how you lost the file to begin with unless the game was unable to write it one day.

**Reply by n00b660** [reply]
That's just the sever settings not setup on the dedicated server, i don't think it's the cause of me not being able to host from the in-game hosting menu and I have plenty of disc space.

**Reply by BlackMoons** [reply]
That file is used by the non dedicated server as well. 

If your error says it can't open the file.. Its likely the cause. It might not be the root cause (Sounds like you don't have write permission to that folder or something?) but it definitely is the cause.

**Reply by n00b660** [reply]
oh snap

**Reply by n00b660** [reply]
I think i just fixed it 
<img width="677" alt="image" src="https://github.com/FakeFishGames/Barotrauma/assets/160653286/269b722f-3c1e-45bb-908e-4630163510e4">

**Reply by n00b660** [reply]
Im gonna try hosting ingame

**Reply by n00b660** [reply]
still crashes...

**Reply by SomeRandomNoobKekeke** [reply]
I guess you'll have to host dedicated server then
Also lua doc says:
![Screenshot_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/3575a545-4e0b-44cd-a9dc-ff7ab8b60c05)
So i guess problem might be with your steam
Dedicated server
![Screenshot_2](https://github.com/FakeFishGames/Barotrauma/assets/122838333/5cf9c52b-77d8-45fe-bcc6-55411e3b4a23)

**Reply by n00b660** [reply]
well maybe, steam is acting up and i can't even launch the game using it.

**Reply by n00b660** [reply]
Solution found: it was all the antivirus, just turned it off and everything is working perfectly fine now! thanks for helping.

**Reply by SomeRandomNoobKekeke** [reply]
![Shrimple](https://github.com/FakeFishGames/Barotrauma/assets/122838333/e7184b6d-417c-48a2-bea7-abea6b915435)
